python-checks/src/main/java/org/sonar/python/checks/hotspots/HardCodedCredentialsEntropyCheck.java

```diff
@@ -0,0 +1,214 @@
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.ASSIGNMENT_STMT, this::checkAssignment);
+    context.registerSyntaxNodeConsumer(Tree.Kind.ANNOTATED_ASSIGNMENT, this::checkAnnotatedAssignment);
+    context.registerSyntaxNodeConsumer(Tree.Kind.PARAMETER_LIST, this::checkParameterList);
+    context.registerSyntaxNodeConsumer(Tree.Kind.REGULAR_ARGUMENT, this::checkRegularArgument);
+    context.registerSyntaxNodeConsumer(Tree.Kind.DICTIONARY_LITERAL, this::checkDictionaryLiteral);
+  }
+
+  private void patternMatch(Name name, Tree location, String value, SubscriptionContext subscriptionContext) {
+    patternMatch(name.name(), location, value, subscriptionContext);
+  }
+
+  private void patternMatch(String name, Tree location, String value, SubscriptionContext subscriptionContext) {
+    if (!valuePassesPostValidation(value) || !entropyShouldRaise(value)) {
+      return;
+    }
+    patterns().stream()
+      .filter(pattern -> pattern.matcher(name).matches())
+      .findFirst()
+      .ifPresent(pattern -> subscriptionContext.addIssue(location, String.format(MESSAGE, name)));
+  }
+
+  private void checkParameterList(SubscriptionContext subscriptionContext) {
+    var parameterList = (ParameterList) subscriptionContext.syntaxNode();
+    parameterList.nonTuple().stream()
+      .filter(parameter -> parameter.name() != null)
+      .filter(parameter -> parameter.defaultValue() != null)
+      .filter(parameter -> parameter.defaultValue().is(Tree.Kind.STRING_LITERAL))
+      .forEach(parameter -> {
+        var value = ((StringLiteral) parameter.defaultValue()).trimmedQuotesValue();
+        patternMatch(parameter.name(), parameter.defaultValue(), value, subscriptionContext);
+      });
+  }
+
+  private void checkDictionaryLiteral(SubscriptionContext subscriptionContext) {
+    var dictionaryLiteral = (DictionaryLiteral) subscriptionContext.syntaxNode();
+    dictionaryLiteral.elements().stream().filter(e -> e.is(Tree.Kind.KEY_VALUE_PAIR))
+      .map(KeyValuePair.class::cast)
+      .filter(keyValuePair -> keyValuePair.value().is(Tree.Kind.STRING_LITERAL))
+      .forEach(keyValuePair -> {
+        var value = ((StringLiteral) keyValuePair.value()).trimmedQuotesValue();
+        var key = keyValuePair.key();
+        if (!key.is(Tree.Kind.STRING_LITERAL)) {
+          return;
+        }
+        patternMatch(((StringLiteral) key).trimmedQuotesValue(), keyValuePair.value(), value, subscriptionContext);
+      });
+  }
+
+  private void checkRegularArgument(SubscriptionContext subscriptionContext) {
+    var regularArgument = (RegularArgument) subscriptionContext.syntaxNode();
+    var keywordArgument = regularArgument.keywordArgument();
+    if (keywordArgument == null) {
+      return;
+    }
+    var expression = regularArgument.expression();
+    if (!expression.is(Tree.Kind.STRING_LITERAL)) {
```

Comment:
Minor: this condition and cast below could be replaced with the `instanceof` condition:
```java
if (regularArgument.expression() instanceof StringLiteral expression) {
      var value = expression.trimmedQuotesValue();
      patternMatch(keywordArgument, regularArgument, value, subscriptionContext);
    }
```